lib/computer_brain.rb

```diff
@@ -1,14 +1,101 @@
 require 'pry'
 
 class ComputerBrain
-  attr_accessor :attacked_keys, :available_keys
+  attr_accessor :attacked_keys, :available_keys, :validated_keys
 
-  def initialize(user_board)
+  def initialize(user_board, cpu_board)
     @user_board = user_board
+    @cpu_board = cpu_board
     @attacked_keys = []
     @available_keys = @user_board.cells.keys
+    @validated_keys = []
   end
 
+  ##Computer place methods
+
+  def make_limited_placement_array(ship)
+    length = ship.length
+    max_width = @cpu_board.width - length + 1
+    max_height = (@cpu_board.height - length + 65).chr
+
+    potential_first_keys = []
+    ("A"..max_height).to_a.each do |letter|
+      (1..max_width).to_a.each do |number|
+        potential_first_keys << "#{letter}#{number}"
+      end
+    end
+    return potential_first_keys
+  end
+
+  def directional_search(coordinate, direction, ship)
+    letter = coordinate.split("")[0]
+    number = coordinate.split("")[1].to_i
+    potential_keys = []
+
+    if direction == :vertical
```

Comment:
Do you think it makes sense to put all of the 'if direction == :vertical' and the 'if direction == :horizontal' code into separate methods for readability? That's what everyone kept telling me to do yesterday when I was getting help on the get_last_hit problem.